hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java

```diff
@@ -37,10 +37,12 @@ public static StructuredTraceGraph buildGraph(StructuredTrace trace) {
 
     if (shouldRebuildTraceEventsGraph || shouldRebuildTraceEntitiesGraph) {
       Instant start = Instant.now();
-      StructuredTraceGraph graph =
-          shouldRebuildTraceEventsGraph
-              ? StructuredTraceGraph.reCreateTraceEventsGraph(trace)
-              : StructuredTraceGraph.reCreateTraceEntitiesGraph(trace);
+      StructuredTraceGraph graph = cachedGraphThreadLocal.get();
```

Comment:
Is there a chance this can be null?

Reply:
Although that case shouldn't arise, but added a check for the cachedGraph in the first check